pkg/materialize/source_mysql.go

```diff
@@ -0,0 +1,74 @@
+package materialize
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type SourceMySQLBuilder struct {
+	Source
+	clusterName     string
+	size            string
+	mysqlConnection IdentifierSchemaStruct
+	tables          []TableStruct
+}
+
+func NewSourceMySQLBuilder(conn *sqlx.DB, obj MaterializeObject) *SourceMySQLBuilder {
+	b := Builder{conn, BaseSource}
+	return &SourceMySQLBuilder{
+		Source: Source{b, obj.Name, obj.SchemaName, obj.DatabaseName},
+	}
+}
+
+func (b *SourceMySQLBuilder) ClusterName(c string) *SourceMySQLBuilder {
+	b.clusterName = c
+	return b
+}
+
+func (b *SourceMySQLBuilder) Size(s string) *SourceMySQLBuilder {
```

Comment:
this isn't relevant anymore, right?

Reply:
This is used by the read func to get the size of the cluster where the source is created in:

https://github.com/MaterializeInc/terraform-provider-materialize/blob/a815ab55a8de5ba48a47d1a8cf5d003e93c7c052/pkg/materialize/source.go#L119

This was part of the migration that allowed us to remove notion of linked clusters in Terraform.